extensions/amp-user-notification/amp-user-notification.md

```diff
@@ -0,0 +1,127 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+### <a name="amp-user-notification"></a> `amp-user-notification`
+
+Displays a dismissable notification to the user. By supplying two URLs that
+get called before the notification is shown and after it is dismissed,
+it is possible to control per user as to whether the notification should
+be shown (using the `ampUserId` value).
+E.g. it could only be shown to users in certain geo locations or
+it could not be shown against after the user has dismissed it once.
+
+---
+
+#### Usage
+
+`amp-user-notification` requires 2 URLs which can be provided by
+the `data-show-if-href` and `data-dismiss-href` attributes. An `id` is required
+as multiple `amp-user-notification` elements are allowed and the
+id is used to differentiate them.
+
+To close `amp-user-notification` add a `on` attribute to a button with the
+following value scheme `on="event:idOfUserNotificationElement.dismiss"`
+(see example below). This user action also triggers the `POST` to the
+`data-dismiss-href` URL.
+
+Example:
+
+```html
+<amp-user-notification
+    layout=container
+    id="amp-user-notification1"
+    data-show-if-href="http://localhost:8000/api/notification"
+    data-dismiss-href="http://localhost:8000/api/notification">
+    This site uses cookies to personalize content.
+    <a href="">Learn more.</a>
+   <button on="tap:amp-user-notification1.dismiss">I accept</button>
+</amp-user-notification>
+```
+
+---
+
+#### Attributes
+
+**data-show-if-href** (Required)
+
+AMP will make a request to this URL to determine whether the
+notification should be shown.
+
+ - `GET request` json fields
+    - `elementId`
+    - `ampUserId`
+
+  Example:
+    ```json
+    { "elementId": "id-of-amp-user-notification", "ampUserId": "ampUserIdString" }
+    ```
+
+ - `GET response` json fields
+    The response must contain a single JSON object with a field
+    "showNotification" of type boolean. If this field is `true` the
+    notification will be shown, otherwise it will not be shown.
+
+    - `showNotification`
+
+    Example:
+    ```json
+    { "showNotification": true }
+    ```
+
+**data-dismiss-href** (Required)
+
+AMP will make a request to this URL to transmit the `elementId` and
+`ampUserId` only when the user has explicitly agreed.
+
+Use the ampUserId field to store that the user has seen the notification before
+if you want to avoid showing it in the future.
+
+  - `POST request` json fields
+
+    - `elementId`
+    - `ampUserId`
+
+    Example:
+    ```json
+    { "elementId": "id-of-amp-user-notification", "ampUserId": "ampUserIdString" }
+    ```
+  - `POST response` should be a 200 HTTP code and no data is expected back.
+
+**credentials**
+
+defines a `credentials` option as specified by the
+[Fetch API](https://fetch.spec.whatwg.org/). To send credentials, pass the
```

Comment:
will do. what did you think about defaulting to always sending the credentials/cookies ?